sys/kern/src/arch/riscv32.rs

```diff
@@ -443,50 +444,43 @@ fn timer_handler() {
     crate::profiling::event_timer_isr_exit();
 }
 
-// NOTE: The kernel currently assumes that only Context 0 handles interrupts.
 pub fn disable_irq(n: u32) {
-    Plic::mask(0, n as usize);
+    let cur_mie = register::mie::read();
+    let new_mie = cur_mie.bits() & !(0x1 << n);
+    unsafe {
+        asm!("
+            csrw mie, {x}",
+            x = in(reg) new_mie,
+        );
```

Comment:
Would `cur_mie.set(new_mie)` work instead of `asm`?

Reply:
There is no general-purpose set function for `mie` in the `riscv` crate, for some reason.

Reply:
Aah. I probably misunderstood what `set` and `clear` mean in [mie.rs](https://github.com/rivosinc/riscv/blob/dev/fawaz/plic/src/register/mie.rs#L74)

Reply:
They do generate functions for it, but they're private. That whole crate is wonky and needs a cleanup.